src/lambda/local/debugConfiguration.ts

```diff
@@ -240,32 +240,34 @@ export function getArchitecture(
     template: CloudFormation.Template | undefined,
     resource: CloudFormation.Resource | undefined,
     invokeTarget: AwsSamDebuggerConfiguration['invokeTarget']
-): Architecture {
+): Architecture | undefined {
     let arch: string | undefined
     let architectureLocation: string
     if (template) {
-        arch = (CloudFormation.getArrayForProperty(resource?.Properties, 'Architectures', template) ?? ['x86_64'])[0]
+        arch = (CloudFormation.getArrayForProperty(resource?.Properties, 'Architectures', template) ?? [])[0]
```

Comment:
No need to coalesce anymore right?

Reply:
Still needed because Typescript _really_ doesn't like `[0]` on something that might be undefined.

Reply:
Ah ok yup it's still an array. GitHub needs to add TypeScript support for PRs